packages/react/src/ReactNoopUpdateQueue.js

```diff
@@ -21,12 +21,15 @@ function warnNoop(publicInstance, callerName) {
     }
     warning(
       false,
-      '%s(...): Can only update a mounted or mounting component. ' +
-        'This usually means you called %s() on an unmounted component. ' +
-        'This is a no-op.\n\nPlease check the code for the %s component.',
-      callerName,
+      "Can't call %s on a component that is not yet mounted. " +
+        'This is a no-op, but it might indicate a bug in your application.\n\n' +
```

Comment:
Can't call %s on a component before mounting it. This is a no-op, but it likely indicates a bug in your application. Instead, assign to \`this.state\` directly or define a \`state = {};\` class property with the desired state.

^ Maybe? It looks a little wordy.

Reply:
Do you mean you want to drop the second part? I thought it's kind of important because these issues will get more prominent with async.

Reply:
I think I see what you mean. If this only affects constructor we might as well not talk about this.